### PR TITLE
#1123 - Update rework

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1141
https://github.com/openvstorage/alba-asdmanager/pull/165
https://github.com/openvstorage/framework-alba-plugin/pull/278

File: ovs/extensions/plugins/asdmanager.py [hunk 1]
```diff
@@ -194,20 +203,43 @@ def delete_asd(self, disk_id, asd_id):
...
+        Retrieve the package information for this ALBA node
         :return: Latest available version and services which require a restart
         """
-        return self._call(requests.get, 'update/information', timeout=120, clean=True)
+        # For backwards compatibility we first attempt to retrieve using the newest API
+        try:
+            return self._call(requests.get, 'update/package_information', timeout=120, clean=True)
+        except NotFoundError:
+            update_info = self._call(requests.get, 'update/information', timeout=120, clean=True)
+            return {'alba': [{'candidate': '',
```
Comment (khenderick) [inline ovs/extensions/plugins/asdmanager.py]: Shouldn't you check whether there's an update for alba available?

File: ovs/extensions/plugins/asdmanager.py [hunk 2]
```diff
@@ -194,20 +203,38 @@ def delete_asd(self, disk_id, asd_id):
...
+            return self._call(requests.get, 'update/package_information', timeout=120, clean=True)
+        except NotFoundError:
+            update_info = self._call(requests.get, 'update/information', timeout=120, clean=True)
+            return {'alba': [{'candidate': update_info['version'],
+                              'installed': update_info['installed'],
+                              'name': 'openvstorage-sdm',
+                              'namespace': 'alba',
+                              'services_to_restart': []}]}
 
     def execute_update(self, status):
```
Comment (khenderick) [inline ovs/extensions/plugins/asdmanager.py]: Why is the status still passed in? It's not relevant anymore for the new logic right? I'd expect that you hardcode ```start``` (or whatever it is to trigger an update on the old sdm) to the old API, so it's triggered, and as soon as it's restarted you'll be able to track progress using the new APIs.

File: ovs/extensions/plugins/asdmanager.py [hunk 3]
```diff
@@ -241,8 +268,11 @@ def list_maintenance_services(self):
...
         :return: dict of services
         """
-        return self._call(requests.get, 'maintenance')['services']
+        return self._call(requests.get, 'maintenance', clean=True)['services']
 
     def _refresh(self):
         self._base_url = 'https://{0}:{1}'.format(self.node.ip, self.node.port)
         self._base_headers = {'Authorization': 'Basic {0}'.format(base64.b64encode('{0}:{1}'.format(self.node.username, self.node.password)).strip())}
+
+    def test(self):
```
Comment (khenderick) [inline ovs/extensions/plugins/asdmanager.py]: Test?

File: ovs/lib/albaupdate.py [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+            for alba_node in AlbaNodeList.get_albanodes():
+                AlbaUpdateController._logger.debug('{0}: Updating SDM'.format(alba_node.ip))
+                try:
+                    result = alba_node.client.execute_update(status=None)
+                    # Backwards compatibility
+                    if result and 'status' in result and result['status'] != 'done':
+                        counter = 0
+                        max_counter = 12
+                        while counter < max_counter:
+                            status = alba_node.client.execute_update(status=None).get('status')
```
Comment (khenderick) [inline ovs/lib/albaupdate.py]: Is this status looping still required? I'd expect that you don't care about the return value in the new  code and your update tracking will be using the new flow?

File: ovs/extensions/plugins/asdmanager.py (hunk 3 above)
Comment (kvanhijf) [inline ovs/extensions/plugins/asdmanager.py]: Yes indeed

**Pull request merged**